# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/d3d/SurfaceD3D.cpp
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/renderer/d3d/SurfaceD3D.cpp
--- a/gfx/angle/checkout/src/libANGLE/renderer/d3d/SurfaceD3D.cpp	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/d3d/SurfaceD3D.cpp	Sat Sep 07 13:38:36 2019 +0000
@@ -106,7 +106,7 @@
     if (mBuftype == EGL_D3D_TEXTURE_ANGLE)
     {
         ANGLE_TRY(mRenderer->getD3DTextureInfo(mState.config, mD3DTexture, &mFixedWidth,
-                                               &mFixedHeight, &mColorFormat));
+                                               &mFixedHeight, nullptr, &mColorFormat));
         if (mState.attributes.contains(EGL_GL_COLORSPACE))
         {
             if (mColorFormat->id != angle::FormatID::R8G8B8A8_TYPELESS &&
@@ -346,7 +346,7 @@
 
 egl::Error SurfaceD3D::swap(const gl::Context *context)
 {
-    DisplayD3D *displayD3D = GetImplAs<DisplayD3D>(context->getCurrentDisplay());
+    DisplayD3D *displayD3D = GetImplAs<DisplayD3D>(context->getDisplay());
     return swapRect(displayD3D, 0, 0, mWidth, mHeight);
 }
 
@@ -356,7 +356,7 @@
                                      EGLint width,
                                      EGLint height)
 {
-    DisplayD3D *displayD3D = GetImplAs<DisplayD3D>(context->getCurrentDisplay());
+    DisplayD3D *displayD3D = GetImplAs<DisplayD3D>(context->getDisplay());
     return swapRect(displayD3D, x, y, width, height);
 }
 